cranelift/filetests/filetests/isa/x64/cmp-mem-bug.clif

```diff
@@ -37,10 +37,9 @@ block0(v0: f64, v1: i64):
 ; nextln: andq    $$1, %rsi
 
     v5 = select.f64 v3, v0, v0
-; nextln: ucomisd %xmm1, %xmm0
+; nextln: ucomisd %xmm0, %xmm1
 ; nextln: movaps  %xmm0, %xmm1
-; nextln: jnp $$next; movsd %xmm0, %xmm1; $$next:
-; nextln: jz $$next; movsd %xmm0, %xmm1; $$next:
```

Comment:
@cfallin, I believe this test is incorrect--both previously and with the current lowering. By first moving `%xmm0` to `%xmm1` with `movaps`, the flag-checking jumps really have no effect (they just move almost the same data again). Thoughts?

Reply:
Zooming out: not only was the `FloatCC::Equal` lowering incorrect for FP, but there are no tests (that I can see) that exercise the other variants.

Reply:
Sounds like we need some runtests for these `select`s, rather than just compile tests.